File: Problems-and-challenges/project-euler/problem005.rb
#!/usr/bin/env ruby

# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
# What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

class Integer
	def divisible_to(x)
		x.downto(1).all? { |e| self % e == 0  }
	end
end

number = 1
while !number.divisible_to(20)
	number += 1
end
puts number

File: project-euler/005/problem005.py
#!/usr/bin/env python
def greatest_common_divisor(a, b):
    while b:
        a, b = b, a % b
    return a


def least_common_multiple(a, b):
    return a * b // greatest_common_divisor(a, b)


def lcm_of_sequence(seq):
    return reduce(least_common_multiple, seq)


# Smallest number evenly divisible by all numbers from 1 to 20.
print lcm_of_sequence(range(1, 20))
